src/monodroid/jni/embedded-assemblies.c

```diff
@@ -388,7 +396,7 @@ gather_bundled_assemblies_from_apk (
 				add_type_mapping (&managed_to_java_maps, apk, cur_entry_name, ((const char*) mmap_info.area) + offset);
 				continue;
 			}
-			if (strncmp ("assemblies/", cur_entry_name, sizeof ("assemblies/")-1) != 0)
+			if (strncmp (assemblies_prefix, cur_entry_name, strlen (assemblies_prefix)-1) != 0)
```

Comment:
I don't think that the `strlen(...)-1` is needed here. `sizeof("foo")` is 4, because it includes the space for the terminating null. `strlen("foo")`, meanwhile, is 3, as **strlen**(3) *doesn't* include the terminating null in the length. I thus believe you only need `strlen (get_assemblies_prefix ())`.